node_modules/oae-principals/lib/test/util.js

```diff
@@ -1047,3 +1125,256 @@ var assertUpdateUserFails = module.exports.assertUpdateUserFails = function(rest
+ * @throws {AssertionError}                     Thrown if the operation is successful
+ */
+var assertResendEmailTokenFails = module.exports.assertResendEmailTokenFails = function(restCtx, userId, httpCode, callback) {
+    RestAPI.User.resendEmailToken(restCtx, userId, function(err) {
+        assert.ok(err);
+        assert.strictEqual(err.code, httpCode);
+        return callback();
+    });
+};
+
+/**
+ * Assert that a user has a pending email token
+ *
+ * @param  {RestContext}    restCtx         The REST context of a user who will perform the request
+ * @param  {String}         userId          The id of the user for which to check whether they have a pending email token
+ * @param  {Function}       callback        Invoked when the check has been performed
+ * @param  {String}         callback.email  The email address for which there is a token
+ * @throws {AssertionError}                 Thrown if the operation is unsuccessful
+ */
+var assertHasPendingEmailTokenSucceeds = module.exports.assertHasPendingEmailTokenSucceeds = function(restCtx, userId, callback) {
+    RestAPI.User.hasPendingEmailToken(restCtx, userId, function(err, data) {
+        assert.ok(!err);
+        return callback(data.email);
+    });
+};
+
+/**
+ * Assert that a user has no pending email token or that the operation fails
+ *
+ * @param  {RestContext}    restCtx         The REST context of a user who will perform the request
+ * @param  {String}         userId          The id of the user for which to check whether they have a pending email token
+ * @param  {Number}         httpCode        The expected HTTP code of the failed has pending email token operation
+ * @param  {Function}       callback        Invoked when the check has been performed
+ * @param  {String}         callback.email  The email address for which there is a token
+ * @throws {AssertionError}                 Thrown if the operation is successful
+ */
+var assertHasPendingEmailTokenFails = module.exports.assertHasPendingEmailTokenFails = function(restCtx, userId, httpCode, callback) {
+    RestAPI.User.hasPendingEmailToken(restCtx, userId, function(err, data) {
+        assert.ok(err);
+        assert.strictEqual(err.code, httpCode);
+        return callback();
+    });
+};
+
+/**
+ * Assert that an email token can be deleted
+ *
+ * @param  {RestContext}    restCtx         The REST context of a user who will perform the request
+ * @param  {String}         userId          The id of the user for which to delete the pending email token
+ * @param  {Function}       callback        Invoked when the delete operation has been performed
+ * @throws {AssertionError}                 Thrown if the operation is unsuccessful
+ */
+var assertDeleteEmailTokenSucceeds = module.exports.assertDeleteEmailTokenSucceeds = function(restCtx, userId, callback) {
+    // Verify the user has a token
+    assertHasPendingEmailTokenSucceeds(restCtx, userId, function() {
```

Comment:
I don't think we need to perform this verification before we delete the email token. Just the one after as the expected aftereffect.